src/main/java/net/goldiriath/plugin/game/item/CustomItemManager.java

```diff
@@ -86,21 +86,35 @@ protected void onStart() {
                 }
             }
 
-            // No validation below this point
-            // Create itemstack
-            final ItemStack stack = new ItemStack(type, 1);
+            // Create UUID and itemstack
+            ItemStack stack = new ItemStack(type, 1);
+
+            // Set data loaded from this config here.
+            stack.addEnchantments(enchantments);
+            stack.getData().setData(data);
 
             // Create and load metadata, if present
             UUID stackUuid = UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8));
             final GItemMeta meta = GItemMeta.createItemMeta(stack, stackUuid);
             plugin.im.getItemMeta().getMetaCache().put(stackUuid, meta);
-            if (section.isConfigurationSection("meta")) {
-                meta.loadFrom(section.getConfigurationSection("meta"));
+
+            // Load metadata from this section
+            // Most importantly: name, level, (armor type), tier, and lore
+            meta.loadFrom(section);
+
+            // Validate armor type
+            if (meta.getArmorType() != null && !InventoryUtil.isArmor(type)) {
+                logger.warning("Skipping item: " + id + ". Armor type specified for non-armor!");
+                continue;
+            }
+            if (meta.getArmorType() == null && InventoryUtil.isArmor(type)) {
```

Comment:
@cashenkes I don't think so. Weapons don't have a heavy/light variant defined right now. This line just ensures that either the type is armor, and we have specified weather its light/heavy armor, or that the type is not armor, and there is no heavy/light defined.